client/components/themenew/theme.jsx

```diff
@@ -0,0 +1,95 @@
+import { isDefaultGlobalStylesVariationSlug, ThemeCard } from '@automattic/design-picker';
+import page from 'page';
+import ThemeTypeBadge from 'calypso/components/theme-type-badge';
+import { decodeEntities } from 'calypso/lib/formatting';
+import { useSelector } from 'calypso/state';
+import { useSiteGlobalStylesStatus } from 'calypso/state/sites/hooks/use-site-global-styles-status';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import {
+	getThemeDetailsUrl,
+	isInstallingTheme,
+	isThemeActive,
+} from 'calypso/state/themes/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import useImageClick from './hooks/use-image-click';
+import useThemeShowcaseTracks from './hooks/use-theme-showcase-tracks';
+import useThemesBookmark from './hooks/use-themes-bookmark';
+import ThemeActions from './theme-actions';
+import { useThemeContext } from './theme-context';
+import ThemePlaceholder from './theme-placeholder';
+import { useThemeShowcaseContext } from './theme-showcase-context';
+import ThemeThumbnail from './theme-thumbnail';
+import ThemeUpdateAlert from './theme-update-alert';
+
+export default function Theme( { isPlaceholder } ) {
+	const { selectedStyleVariation, setSelectedStyleVariation, theme, themeId } = useThemeContext();
+	const { tabFilter } = useThemeShowcaseContext();
+
+	const siteId = useSelector( getSelectedSiteId );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const themeDetailsUrl = useSelector( ( state ) =>
+		getThemeDetailsUrl( state, themeId, siteId, {
+			styleVariationSlug: selectedStyleVariation?.slug,
+			tabFilter,
+		} )
+	);
+	const isActive = useSelector( ( state ) => isThemeActive( state, themeId, siteId ) );
+	const isInstalling = useSelector( ( state ) => isInstallingTheme( state, themeId, siteId ) );
+
+	const { imageClickUrl, imageLabel, onImageClick } = useImageClick( { selectedStyleVariation } );
+	const { shouldLimitGlobalStyles } = useSiteGlobalStylesStatus( siteId );
```

Comment:
Every theme is going to be fetching this global state and subscribing to changes. Since this is site-wide value maybe we could find a more efficient way than retrieving it on every theme card (Context or resolving it on on the higher component instead).

Reply:
This is a good candidate for the Showcase context indeed! 👍